pkg/prom/instance/instance_integration_test.go

```diff
@@ -0,0 +1,196 @@
+package instance
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cortexproject/cortex/pkg/util/test"
+	"github.com/go-kit/kit/log"
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/prometheus/config"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+)
+
+// TestInstance_Update performs a full integration test by doing the following:
+//
+// 1. Launching an HTTP server which can be scraped and also mocks the remote_write
+//    endpoint.
+// 2. Creating an instance config with no scrape_configs or remote_write configs.
+// 3. Updates the instance with a scrape_config and remote_write.
+// 4. Validates that after 15 seconds, the scrape endpoint and remote_write
+//    endpoint has been called.
+func TestInstance_Update(t *testing.T) {
+	prometheus.DefaultRegisterer = prometheus.NewRegistry()
```

Comment:
I'm not happy that I have to do this, and I'm planning on cleaning this up with the PR for #141.